File: Typescript/Tree/Node.ts
import { Comparable } from "./BinarySearchTree";

class NODE<T extends Comparable<T> | null> {
  constructor(
    public data: T,
    public left: NODE<T> | null = null,
    public right: NODE<T> | null = null
  ) {}

  getData(): T | null {
    return this.data;
  }

  setData(data: T): void {
    this.data = data;
  }

  getRight(): NODE<T> | null {
    return this.right;
  }

  setRight(right: NODE<T> | null): void {
    this.right = right;
  }

  getLeft(): NODE<T> | null {
    return this.left;
  }

  setLeft(left: NODE<T> | null): void {
    this.left = left;
  }
}

export default NODE;

File: Java/src/Tree/Node.java
package Tree;

public class Node<T extends Comparable<T>> {
    private T data;
    private Node<T> left;
    private Node<T> right;

    public Node(Node<T> right, T data, Node<T> left) {
        this.right = right;
        this.data = data;
        this.left = left;
    }

    ;

    public Node<T> getLeft() {
        return left;
    }

    public void setLeft(Node<T> left) {
        this.left = left;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public Node<T> getRight() {
        return right;
    }

    public void setRight(Node<T> right) {
        this.right = right;
    }

    @Override
    public String toString() {
        return data.toString();
    }
}
